CopilotKit/packages/backend/src/lib/copilotkit-backend-implementation.ts

```diff
@@ -89,25 +89,18 @@ export class CopilotBackendImplementation {
       }
     }
 
-    // merge server side functions with langserve functions
-    let mergedTools = mergeServerSideTools(
-      this.actions.map(actionToChatCompletionFunction),
-      langserveFunctions.map(actionToChatCompletionFunction),
-    );
-
-    // merge with client side functions
-    mergedTools = mergeServerSideTools(mergedTools, forwardedProps.tools);
+    const serversideTools: Action<any>[] = [...this.actions, ...langserveFunctions];
+    const mergedTools = flattenToolCallsNoDuplicates([
```

Comment:
Previously we were calling the old `mergeServerSideTools` twice, once with client tools.

We were simply using it as a de-duplication step by precedence as prioirty.

So i re-implemented `flattenToolCallsNoDuplicates` to take an array of tools and de-dupe it (earlier = higher priority).